lib/Core/SQLiteBuildDB.cpp

```diff
@@ -310,7 +323,11 @@ class SQLiteBuildDB : public BuildDB {
 
   virtual uint64_t getCurrentIteration(bool* success_out, std::string *error_out) override {
     std::lock_guard<std::mutex> guard(dbMutex);
-    assert(db);
+
+    if (!open(error_out)) {
+      *success_out = false;
+      return 0;
+    }
```

Comment:
This works for now as a minimal fix, but it feels unfortunately we have to have the lazy behavior versus a more explicit open/close that could be propagated up to the `BuildEngine` -- did you explore how much work it would be to make this more explicit in the BuildDB API?

Reply:
For example, if the engine ever decided to call this but never call build, then the db wouldn't be closed.

Reply:
As discussed offline, yeah, this is a bit awkward and could definitely be helped with a more explicit API, such as a build context.  This is indeed intended to be a minimal fix for the time being.  Tracking the refactor in rdar://problem/42310556